src/app/Http/Middleware/CheckSubscriptionPermission.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use Closure;
+use Illuminate\Support\Facades\Auth;
+use App\Providers\RouteServiceProvider;
+
+class CheckSubscriptionPermission
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        // get the presentation presenter
+        $presenter = $request->presentation->users()->where('role', 'presenter')->first();
```

Comment:
what if there are more than one presenter ? or if we implement other roles than presenter that should be able to manage presentations ?